File: Exercism Codes/Java/WordCount.java
import java.util.Map;
import java.util.HashMap;

class WordCount {
    public Map<String, Integer> phrase(String input) {
        String[] substring = input.toLowerCase().trim().split("[^a-zA-Z0-9']+");
        Map<String,Integer> map = new HashMap<String,Integer>();
        for (String i : substring) {
            String j = i.replaceAll("^'|'$", "");
            if(j != ""){
                if(!map.containsKey(j)){
                    map.put(j, 1);
                }else {
                    map.put(j, map.get(j) + 1);
                }
            }
        }
        return map;
    }
}

File: Exercism Codes/C#/WordCount.cs
using System.Text.RegularExpressions;

public static class WordCount
{
    public static IDictionary<string, int> CountWords(string phrase)
    {
        string[] arr = Regex.Split(phrase.ToLower().Trim(), "[^a-zA-Z0-9']+");
        IDictionary<string, int> result = new Dictionary<string, int>();
        foreach (string word in arr)
        {
            string w = Regex.Replace(word, "^'{1,2}|'{1,2}$", "");
            if (w.Length > 0)
            {
                if (result.ContainsKey(w))
                {
                    result[w]++;
                }
                else
                {
                    result[w] = 1;
                }
            }
        }

        return result;
    }
}